packages/core/contracts/cross-chain-oracle/chain-adapters/Arbitrum_ParentMessenger.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../insured-bridge/avm/Arbitrum_CrossDomainEnabled.sol";
+import "../interfaces/ParentMessengerInterface.sol";
+import "../interfaces/ParentMessengerConsumerInterface.sol";
+import "./ParentMessengerBase.sol";
+import "../../common/implementation/Lockable.sol";
+
+/**
+ * @notice Sends cross chain messages from Ethereum L1 to Arbitrum L2 network.
+ * @dev This contract is ownable and should be owned by the DVM governor.
+ */
+contract Arbitrum_ParentMessenger is
+    Arbitrum_CrossDomainEnabled,
+    ParentMessengerInterface,
+    ParentMessengerBase,
+    Lockable
+{
+    event SetDefaultGasLimit(uint32 newDefaultGasLimit);
+    event SetDefaultMaxSubmissionCost(uint256 newMaxSubmissionCost);
+    event SetDefaultGasPrice(uint256 newDefaultGasPrice);
+    event MessageSentToChild(
+        bytes data,
+        address indexed targetContract,
+        uint256 l1CallValue,
+        uint32 gasLimit,
+        uint256 gasPrice,
+        uint256 maxSubmissionCost,
+        address indexed childAddress,
+        uint256 sequenceNumber
+    );
+    event MessageReceivedFromChild(bytes data, address indexed childAddress, address indexed targetHub);
+
+    // TODO: Try to read these from Arbitrum system contracts.
+    uint32 public defaultGasLimit = 5_000_000;
```

Comment:
We should look into how we can dynamically read this but i wanted to test the contracts for now on test net